neutone_sdk/parameter.py

```diff
@@ -36,26 +48,74 @@ def to_metadata_dict(self) -> Dict[str, str]:
         }
 
 
-class KnobNeutoneParameter(NeutoneParameter):
+class ContinuousNeutoneParameter(NeutoneParameter):
     """
-    Defines a knob Neutone Parameter that the user can use to control a model.
+    Defines a continuous Neutone Parameter that the user can use to control a model.
 
     The name and the description of the parameter will be shown as a tooltip
-    within the UI. `default_value` must be between 0 and 1 and will be used
-    as a default in the plugin when no presets are available.
+    within the UI.
+    `default_value` must be between 0 and 1 and will be used as a default in the plugin
+    when no presets are available.
+
+    The continuous parameter is considered to be a knob parameter.
     """
     def __init__(self,
                  name: str,
                  description: str,
                  default_value: float,
                  used: bool = True):
         super().__init__(name, description, used)
+        assert 0.0 <= default_value <= 1.0, \
+            "`default_value` for continuous params must be between 0 and 1"
+        self.default_value = default_value
+        self.type = NeutoneParameterType.CONTINUOUS
+
+    def to_metadata_dict(self) -> Dict[str, str]:
+        """Returns a string dictionary containing the metadata of the parameter."""
+        data = super().to_metadata_dict()
+        data["default_value"] = str(self.default_value)
+        return data
+
+
+class CategoricalNeutoneParameter(NeutoneParameter):
+    """
+    Defines a categorical Neutone Parameter that the user can use to control a model.
+
+    The name and the description of the parameter will be shown as a tooltip
+    within the UI.
+    `n_values` must be an int greater than or equal to 2 and less than or equal to 32.
+    `default_value` must be in the range [0, `n_values` - 1].
+    `labels` is a list of strings that will be used as the labels for the parameter.
+
+    The categorical parameter is considered to be a knob parameter.
+    """
+    def __init__(self,
+                 name: str,
+                 description: str,
+                 n_values: int,
+                 default_value: int,
+                 labels: Optional[List[str]] = None,
+                 used: bool = True):
+        super().__init__(name, description, used)
+        assert 2 <= n_values <= constants.MAX_N_CATEGORICAL_VALUES, \
+            "`n_values` for categorical params must between 2 and 20"
+        assert 0 <= default_value <= n_values - 1, \
+            "`default_value` for categorical params must be between 0 and `n_values`-1"
+        self.n_values = n_values
         self.default_value = default_value
-        self.type = NeutoneParameterType.KNOB
+        if labels is None:
+            labels = [str(idx) for idx in range(n_values)]
+        else:
+            assert len(labels) == self.n_values, "labels must have `n_values` elements"
```

Comment:
I think we should also have a maximum number of characters defined here or in the metadata checking file.